lib/src/config.rs

```diff
@@ -128,6 +131,11 @@ impl ConfigBuilder {
         self
     }
 
+    pub fn with_client_certificate_provider(mut self, provider: Box<dyn ClientCertificateProvider>) -> Self {
+        self.client_certificate_provider = Some(provider);
+        self
+    }
+
```

Comment:
when the trait is removed, this could either be

```suggestion
    pub fn with_client_certificate(mut self, client_cert: impl AsRef<Path>) -> Self {
        self.client_certificate = Some(ClientCertificate::new(client_cert));
        self
    }
```

or with a From impl


```suggestion
    pub fn with_client_certificate(mut self, client_cert: impl Into<ClientCertificate>) -> Self {
        self.client_certificate = Some(client_cert.into());
        self
    }
```